chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -230,22 +222,9 @@ pub(crate) fn send_chunk_endorsement_to_block_producers(
     let protocol_version = epoch_manager.get_epoch_protocol_version(&epoch_id).unwrap();
     let endorsement = ChunkEndorsement::new(epoch_id, chunk_header, signer, protocol_version);
     for block_producer in block_producers {
-        if signer.validator_id() == &block_producer {
-            // Our own endorsements are not always valid (see issue #11750).
-            if let Err(err) = chunk_endorsement_tracker
```

Comment:
Just for my own understanding; is the removal of the mutex around the tracker enabled by this change?

Reply:
Yes, so we needed the mutex before for inner mutability pattern. We couldn't pass a mut ref to send_chunk_endorsement_to_block_producers